borsh-derive/src/internals/schema/enums/snapshots/complex_enum.snap

```diff
@@ -31,16 +23,17 @@ impl borsh::BorshSchema for A {
             wrapper: Wrapper,
             filling: Filling,
         }
-        <ABacon as borsh::BorshSchema>::add_definitions_recursively(definitions);
-        <AEggs as borsh::BorshSchema>::add_definitions_recursively(definitions);
+        <() as borsh::BorshSchema>::add_definitions_recursively(definitions);
+        <() as borsh::BorshSchema>::add_definitions_recursively(definitions);
         <ASalad as borsh::BorshSchema>::add_definitions_recursively(definitions);
         <ASausage as borsh::BorshSchema>::add_definitions_recursively(definitions);
         let definition = borsh::schema::Definition::Enum {
             tag_width: 1,
             variants: borsh::__private::maybestd::vec![
-                ("Bacon".to_string(), < ABacon > ::declaration()), ("Eggs".to_string(), <
```

Comment:
actually, this change implies, that `"nil"` starts to mean different things in different contexts.
It adds a new rule for interpreting `"nil"` as a unit struct variant in certain situations.
Let's keep it more verbose (and more explicit) as it is now. One looks at a variant definition, and sees that it's a unit struct.

Reply:
But with this change one can look at a variant definition and sees it’s a unit struct.  In fact, it’s now easier to see it’s a unit struct since when I see `("Bacon", "nil")` I immediately know it’s a variant with no data; when I see `("Bacon", "ABecon")` I know have to look up what `ABecon` is.

What’s more, `nil` can be mapped to an empty tuple which would better match that it’s used for `()`.  (To be honest, I’d actually change the declaration from `"nil"` to `"()"` and then also change declaration for tuples to use `(...)`).

Reply:
@mina86 i think, it's a good idea to change declarations:
1. `nil` -> `()`
2. `Tuple<T0, T1, T2>` -> `(T0, T1, T2)`
3. `Array<T0, N>` -> `[T0; N]`

i've created #231, please post there if you want to go ahead and create one or more of 3 subtasks prs, 
so that we don't accidentally concurrently do the same thing.